src/components/core/GeoChart.tsx

```diff
@@ -41,31 +59,51 @@ export type IGeoChartInnerProps = ICoreGeoChartProps &
     IDataSourceProviderInjectedProps &
     WrappedComponentProps;
 
+export interface IGeoChartInnerState {
+    legendItemsEnabled: IPushpinCategoryLegendItem[];
+}
+export interface IGeoChartInnerOptions {
+    geoData: IGeoData;
+    segmentData: IPushpinCategoryLegendItem[];
+    sizeData: number[];
+}
 /**
  * Geo Chart react component
  */
-export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, {}> {
+export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, IGeoChartInnerState> {
     public static defaultProps: Partial<IGeoChartInnerProps> = {
         ...commonDefaultProps,
         chartRenderer: renderChart,
         legendRenderer: renderLegend,
-        legendPostion: TOP,
+        legendPosition: TOP,
         config: {
             mapboxAccessToken: "",
         },
     };
-
-    public componentDidUpdate(prevProps: IGeoChartInnerProps) {
-        if (!isEqual(this.props.execution, prevProps.execution)) {
-            this.validateData();
-        }
-    }
-
-    public componentDidMount() {
-        this.validateData();
-    }
+    private geoChartOptions: IGeoChartInnerOptions;
 
     public renderVisualization() {
+        if (this.props.execution) {
+            const {
+                config: { mdObject: { buckets = [] } = {} },
+                execution: { executionResponse },
+                onDataTooLarge,
+            } = this.props;
+
+            const geoData: IGeoData = getGeoData(buckets, executionResponse.dimensions);
+            const validateResult = this.validateData(geoData);
```

Comment:
it should be
```
const {
   config: { mdObject: { buckets = [] } = {} },
   execution: { executionResponse: { dimensions },
   onDataTooLarge,
} = this.props;
const geoData: IGeoData = getGeoData(buckets, dimensions);
const { isDataTooLarge } = this.validateData(geoData);
```

Reply:
It would be better to move validation to here `./src/components/core/base/GeoValidatorHOC.tsx`
Please help me check that it's possible